### PR TITLE
Add Embed Support

### PR DESCRIPTION
**Pull request by tooxo**
Not in a useable state yet. View this as very early stage, updates will follow.

### CODE REVIEW

**Comment by tooxo** [comment]
Tests should now perform correctly.
Test-Prove Image

**Comment by JakeCover** [comment]
Yeah, I'll work out what's wrong with them and then get this PR in. Thanks a ton for your contribution! If you don't mind documenting how it works, that would be fantastic!

**Comment by JakeCover** [comment]
Update: Travis won't let PRs from forks use encrypted variables, which is how I store the bot credentials. I'll see if it will let me run it and be ok, but if you know how to make this work let me know.
image

**Comment by tooxo** [comment]
> If you don't mind documenting how it works, that would be fantastic!

Is there a specific way you want it to be documented? Full Text, Docstring etc?

**Comment by tooxo** [comment]
Added Support for Image, Thumbnail, Video and Author attributes. Tests should also work accordingly.

**Comment by tooxo** [comment]
You can also look at my travis build, which builds correctly.
https://travis-ci.com/tooxo/distest/builds/128603178

**Comment by JakeCover** [comment]
> > If you don't mind documenting how it works, that would be fantastic!
> 
> Is there a specific way you want it to be documented? Full Text, Docstring etc?

If you could document the methods in the docstring, that would be great, I'll plug it into sphinx once you're done. Just look at one of the nearby functions for an example, doesn't need to be anything complicated. Full documentation [here](https://distest.readthedocs.io/en/feature-add_documentation/index.html) in case you want to take a look.

-----

> You can also look at my travis build, which builds correctly.
> https://travis-ci.com/tooxo/distest/builds/128603178

Sound good, once the documentation is done, I'll approve the PR.

File: distest/interface.py [hunk 1]
```diff
@@ -152,6 +152,86 @@ def checkReaction(reaction, user):
...
+        self,
+        message: discord.Message,
+        matches: discord.Embed,
+        attributes_to_prove: list = None,
+    ):
+        """
+        If ``matches`` doesn't match the embed of ``message``, fail the test.
+        :param message: original message
+        :param matches: embed object to compare to
+        :param attributes_to_prove: a string list with the attributes of the embed, which are to compare
```
Comment (JakeCover) [inline distest/interface.py]: Looks good, just plz explain `attributes_to_prove` a bit better

File: distest/interface.py [hunk 2]
```diff
@@ -242,6 +330,7 @@ def checkReaction(reaction, user):
         :returns: The reply.
         :rtype: discord.Message
         :raises: ResponseDidNotMatchError
+        develop
```
Comment (JakeCover) [inline distest/interface.py]: Whoops, left this in here from a merge conflict. 
```suggestion
```

File: distest/interface.py (hunk 1 above)
Comment (tooxo) [reply distest/interface.py]: Should be done.

**Comment by rade0** [comment]
Hey! This is an awesome and much needed feature, is there anything I can do to help get it over the line?

**Comment by rade0** [comment]
@tooxo @JakeCover Bump :)

**Comment by tooxo** [comment]
@rade0 Thank you for your patience, appreciate it. Didn't touch this repo for a long time, but I remember stopping working on it, because I wasn't entirely sure how to structure the feature. 

For now its, I think, still working, and if you want to compare an embed you can compare all of it or just some attributes.

I don't have any intentions in working on it anymore, I instead implemented a feature in my bot, that, for tests, the bot doesn't use embeds, but thats obviously not ideal.
If you want to take over from here, I can add you as a contributer on my fork and help you getting started.

**Comment by JakeCover** [comment]
> @tooxo @JakeCover Bump :)

I should be working on this more soon, I just got really busy but now thanks to covid19 I have more time! If you want to help in some way, feel free to help out and take tooxo up on their offer on contributions, I'll take a look at this probably today or tomorrow.

**Comment by rade0** [comment]
Awesome thank you both! @JakeCover I'll wait for you to give it the once over and if anything needs doing I'm your man 😀

**Comment by JakeCover** [comment]
Just tested it, looks good! Now I need to figure out how to get the bot tokens into travis on forks, sounds like a problem for future me!

**Comment by rade0** [comment]
Nice one! Will give it a whirl. 

(Don’t worry I also rely on my own future me quite a lot 😂)

**Comment by rade0** [comment]
Just to confirm it works a charm, thanks dude!

**Pull request merged**